packages/playwright-core/src/client/network.ts

```diff
@@ -351,6 +364,8 @@ export class Route extends ChannelOwner<channels.RouteChannel> implements api.Ro
       headers[header.toLowerCase()] = String(headersOption![header]);
     if (options.contentType)
       headers['content-type'] = String(options.contentType);
+    if (options.json)
```

Comment:
If I explicitly pass `contentType` option, we should not override with `application/json`.

Reply:
I've lost 'else', the order war ok :)